Genome/Source/BackingDataType.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  BackingDataType.swift
+//  Genome
+//
+//  Created by Logan Wright on 2/10/16.
+//  Copyright © 2016 lowriDevs. All rights reserved.
+//
+
+import Foundation
+
+extension Node: BackingDataType {
+    public init(_ node: Node) {
+        self = node
+    }
+}
+
+public protocol BackingDataType {
+    var isNull: Bool { get }
+    var boolValue: Bool? { get }
+    var doubleValue: Double? { get }
+    var stringValue: String? { get }
+    var arrayValue: [Self]? { get }
+    var objectValue: [String : Self]? { get }
+    
+    init(_ node: Node)
+}
+
+extension BackingDataType {
+    public var floatValue: Float? {
+        guard let double = doubleValue else { return nil }
+        return Float(double)
+    }
+    
+    public var intValue: Int? {
+        guard let double = doubleValue where double % 1 == 0 else {
+            return nil
+        }
+        
+        return Int(double)
+    }
+    
+    public var uintValue: UInt? {
+        guard let int = intValue where int >= 0 else { return nil }
+        return UInt(int)
+    }
+}
+
+extension Node {
+    public init<T: BackingDataType>(_ data: T) {
+        if let string = data.stringValue {
+            self = .StringValue(string)
+        } else if let bool = data.boolValue {
+            self = .BooleanValue(bool)
+        } else if let number = data.doubleValue {
+            self = .NumberValue(number)
+        } else if let array = data.arrayValue {
+            self = .ArrayValue(array.map(Node.init))
+        } else if let object = data.objectValue {
+            var mutable: [String : Node] = [:]
+            object.forEach { key, val in
+                mutable[key] = Node(val)
+            }
+            self = .ObjectValue(mutable)
+        } else if data.isNull {
+            self = Node.NullValue
+        } else {
+            fatalError("Unable to convert data: \(data)")
+        }
+    }
+}
+
+extension Node {
+    public func dataRepresentation<T: BackingDataType>() -> T {
```

Comment:
Should this and the above also live somewhere else?  Perhaps they should all go in a new file `Node+BackingDataType`